account_payment_mode/models/account_payment_mode.py

```diff
@@ -0,0 +1,123 @@
+    # I dropped sale_ok and purchase_ok fields, because it is replaced by
+    # payment_type = 'inbound' or 'outbound'
+    # In fact, with the new v9 datamodel, you MUST create 2 payment modes
+    # for wire transfer : one for wire transfer from your customers (inbound)
+    # and one for wire transfer to your suppliers (outbound)
+    note = fields.Text(string="Note", translate=True)
+
+    @api.onchange('company_id')
+    def _onchange_company_id(self):
+        for mode in self:
+            mode.variable_journal_ids = False
+            mode.fixed_journal_id = False
+
+    @api.constrains(
+        'bank_account_link', 'fixed_journal_id', 'payment_method_id')
+    def bank_account_link_constrains(self):
+        for mode in self:
+            if mode.bank_account_link == 'fixed':
+                if not mode.fixed_journal_id:
+                    raise ValidationError(_(
+                        "On the payment mode '%s', the bank account link is "
+                        "'Fixed' but the fixed bank journal is not set")
+                        % mode.name)
+                else:
+                    if mode.payment_method_id.payment_type == 'outbound':
+                        if (
+                                mode.payment_method_id.id not in
+                                mode.fixed_journal_id.
+                                outbound_payment_method_ids.ids):
+                            raise ValidationError(_(
+                                "On the payment mode '%s', the payment method "
+                                "is '%s', but this payment method is not part "
+                                "of the payment methods of the fixed bank "
+                                "journal '%s'") % (
+                                    mode.name,
+                                    mode.payment_method_id.name,
+                                    mode.fixed_journal_id.name))
+                    else:
+                        if (
+                                mode.payment_method_id.id not in
+                                mode.fixed_journal_id.
+                                inbound_payment_method_ids.ids):
+                            raise ValidationError(_(
+                                "On the payment mode '%s', the payment method "
+                                "is '%s' (it is in fact a debit method), "
+                                "but this debit method is not part "
+                                "of the debit methods of the fixed bank "
+                                "journal '%s'") % (
+                                    mode.name,
+                                    mode.payment_method_id.name,
+                                    mode.fixed_journal_id.name))
+
+    @api.constrains('company_id', 'fixed_journal_id')
+    def company_id_fixed_journal_id_constrains(self):
+        for mode in self:
+            if mode.fixed_journal_id and mode.company_id != \
```

Comment:
I would put this if up
for mode in self.filtered(
            lambda x: x.fixed_journal_id and x.company_id != x.fixed_journal_id.company_id
        ):